RecoHI/HiTracking/python/HIPixelAdaptiveVertex_cfi.py

```diff
@@ -1,37 +1,32 @@
 import FWCore.ParameterSet.Config as cms
+import RecoVertex.PrimaryVertexProducer.primaryVertexProducer_cfi as _mod
 
-hiPixelAdaptiveVertex = cms.EDProducer("PrimaryVertexProducer",
-    verbose = cms.untracked.bool(False),
-    TkFilterParameters = cms.PSet(
-        algorithm = cms.string('filterWithThreshold'),
-        maxNormalizedChi2 = cms.double(5.0),
-        minSiliconLayersWithHits = cms.int32(0), ## >=0 (was 5 for generalTracks)
-        minPixelLayersWithHits = cms.int32(2),   ## >=2 (was 2 for generalTracks)
-        maxD0Significance = cms.double(3.0),     ## keep most primary tracks (was 5.0)
-        minPt = cms.double(0.0),                 ## better for softish events
-        maxEta = cms.double(100.),
-        trackQuality = cms.string("any"),
-        numTracksThreshold = cms.int32(2)
+hiPixelAdaptiveVertex = _mod.primaryVertexProducer.clone(
+    verbose = False,
+    TkFilterParameters = dict(
+        algorithm = 'filterWithThreshold',
+        maxNormalizedChi2 = 5.0,
+        minSiliconLayersWithHits = 0, ## >=0 (was 5 for generalTracks)
+        minPixelLayersWithHits = 2,   ## >=2 (was 2 for generalTracks)
+        maxD0Significance = 3.0,     ## keep most primary tracks (was 5.0)
+        minPt = 0.0,                 ## better for softish events
+        maxEta = 100.,
+        numTracksThreshold = 2
     ),
-    beamSpotLabel = cms.InputTag("offlineBeamSpot"),
     # label of tracks to be used
-    TrackLabel = cms.InputTag("hiSelectedProtoTracks"),
+    TrackLabel = "hiSelectedProtoTracks",
     # clustering
-    TkClusParameters = cms.PSet(
-        algorithm = cms.string("gap"),
+    TkClusParameters = dict(
+        algorithm = "gap",
         TkGapClusParameters = cms.PSet(
-            zSeparation = cms.double(1.0)        ## 1 cm max separation between clusters
+            zSeparation = cms.double(1.0)       ## 1 cm max separation between clusters
         )
     ),
-    vertexCollections = cms.VPSet(
-      cms.PSet(
-        label = cms.string(''),
-        chi2cutoff = cms.double(3.0),
-        algorithm = cms.string('AdaptiveVertexFitter'),
-        useBeamConstraint = cms.bool(False),
-        maxDistanceToBeam = cms.double(0.1),
-        minNdof  = cms.double(0.0)
+    vertexCollections = {
```

Comment:
The `vertexCollections` VPSet has two PSets in it, the second one being labeled "withBS"
I suspect that as such you clone the VPSet with both PSet's, modify the first one and let the second one unchanged, while in the previous config only one PSet was in: could you please verify with a dump?

Reply:
@perrotta Thanks for your comment. Yes, you are right. 
I have to modify this file to use VPSet and PSet as it was. 
The comparison results are :
```
edmParameterSetDump updated step2.py (wf 140.53)

 vertexCollections: VPSet tracked = ({
    {
      algorithm: string tracked  = 'AdaptiveVertexFitter'
      chi2cutoff: double tracked  = 3
      label: string tracked  = ''
      maxDistanceToBeam: double tracked  = 0.1
      minNdof: double tracked  = 0
      useBeamConstraint: bool tracked  = false
    },
    {
      algorithm: string tracked  = 'AdaptiveVertexFitter'
      chi2cutoff: double tracked  = 2.5
      label: string tracked  = 'WithBS'
      maxDistanceToBeam: double tracked  = 1
      minNdof: double tracked  = 2
      useBeamConstraint: bool tracked  = true
    }
    })
  })
```
```
edmParameterSetDump original step2.py (wf 140.53)

  vertexCollections: VPSet tracked = ({
    {
      algorithm: string tracked  = 'AdaptiveVertexFitter'
      chi2cutoff: double tracked  = 3
      label: string tracked  = ''
      maxDistanceToBeam: double tracked  = 0.1
      minNdof: double tracked  = 0
      useBeamConstraint: bool tracked  = false
    }
    })
  })
```